### PR TITLE
Add social media icons in the right sidebar

### PR DESCRIPTION
**Pull request by dhruvkb**
This pull request adds social media sharing capability to CC Search, similar to the feature for lists already present on the platform. This allows people to share individual images found on the site to 3 popular platforms.

This PR supersedes inactive PR #200 which itself addresses issue #104 raised by @sebworks. In addition to plainly adding the social share buttons, this also corrects the colours used in the buttons to the actual brand colours of the platforms. This also reduces the mess of sidebar sections, caused by using semantically incorrect classes, and reduces SCSS repetition by generalising the sections, while keeping the outer result largely unchanged so as to not disrupt too much.

image

### CODE REVIEW

File: src/components/SocialShareButtons.vue [hunk 1]
```diff
@@ -51,9 +40,9 @@ export default {
   $social-button-transition: all 0.5s ease;
   $social-button-margin: 0.25rem;
 
-  $social-brand-facebook: #3b5998;
-  $social-brand-twitter: #55acee;
-  $social-brand-pinterest: #c32aa3;
+  $social-brand-facebook: #4267b2;
+  $social-brand-twitter: #1da1f2;
+  $social-brand-pinterest: #e60023;
```
Comment (brenoferreira) [inline src/components/SocialShareButtons.vue]: Any particular reason for the color changes?

Comment (dhruvkb) [reply src/components/SocialShareButtons.vue]: The color of Pinterest was very pink-like and did not match the color on the standard 'Pin it' button. So I investigated and found that none of the buttons were the original colors of the brands they represented so I updated them to the official branding colors.

**Comment by dhruvkb** [comment]
@brenoferreira Ok, I will add some tests for these changes at the earliest.

**Comment by dhruvkb** [comment]
@brenoferreira I added the tests for the components in check. Kindly re-review.

**Comment by dhruvkb** [comment]
Brought this file in sync with the `master` branch, resolving conflicts introduced in 44decac.

**Comment by dhruvkb** [comment]
@brenoferreira I've brought this in sync with master. Could you please find some time to re-review this?

**Comment by kgodey** [comment]
Dismissed @brenoferreira's earlier review so this shows up as review requested again.

**Pull request merged**